aspnet/mvc/models/file-uploads.rst

```diff
@@ -0,0 +1,197 @@
+File Uploads
+============
+
+By `Steve Smith`_
+
+ASP.NET MVC actions support uploading of one or more files, using simple model binding for smaller files, or streaming for larger files.
```

Comment:
FYI: Any single buffered file exceeding 64KB will be moved from RAM to a temp file on disk.

There's no magic answer here, it's a resources math problem: RAM & disk vs size & number of concurrent uploads.
 - 5 users uploading 5 1MB files each requires 125MB total temp disk space and up to 320kb RAM.
 - 1 user uploading 1 125MB file requires 125MB temp disk space and up to 64kb RAM.
 - 2000 users uploading 1 64KB file each requires 125MB RAM and no disk.
 - 1 user uploading 2000 64KB files requires 125MB RAM and no disk.

Keep in mind that the server has no way of knowing in advance how many files there are or how big they are until it reads them one at a time. You can use client side verification and authenticated actions to limit it externally.